GVFS/FastFetch/NativeMethods.cs

```diff
@@ -0,0 +1,35 @@
+﻿using GVFS.Common.Tracing;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+
+namespace FastFetch
+{
+    internal static class NativeMethods
+    {
+        public static bool isUnixOS = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+        public static unsafe void WriteFile(ITracer tracer, byte* originalData, long originalSize, string destination, ushort mode)
+        {
+            if (!isUnixOS)
+            {
+                NativeWindowsMethods.WriteFile(tracer, originalData, originalSize, destination, mode);
+            }
+            else
+            {
+                NativeUnixMethods.WriteFile(tracer, originalData, originalSize, destination, mode);
+            }
+        }
+
+        public static bool StatAndUpdateIndexForFile(string path, MemoryMappedViewAccessor indexView, long offset)

```

Comment:
Naming suggestion: `StatAndUpdateIndexForFile` -> `TryStatFileAndUpdateIndex` (since we're not stating the index).

(This is assuming the return value indicates success)